src/bepasty/utils/_compat.py

```diff
@@ -14,9 +15,15 @@
     range_type = xrange  # noqa: F821
     bytes_type = str
     iteritems = lambda d: d.iteritems()  # noqa: E731
+
+    def base64_b64decode(data):
+        return base64.b64decode(data)
 else:
     from urllib.parse import urlparse, urljoin
 
     range_type = range
     bytes_type = bytes
     iteritems = lambda d: iter(d.items())  # noqa: E731
+
+    def base64_b64decode(data):
+        return base64.b64decode(data, validate=True)
```

Comment:
Maybe this is well-meant (like "validating is always better than not"), but in the end, this implements different behaviour for python 2.7 and python 3.x. I don't think we want different behaviour.

So, there should be a ticket about changing this to validating **after** we dropy python 2 support, but this should not be changed within this PR.

Reply:
Hm, maybe no. in my understand, python 2.7 doesn't have validate parameter and validate by default. but python3 default is no validation.

So this sets validate to make be same behavior.

Reply:
no, 2.7 does not validate by default, it strips invalid stuff, like default in 3.x - see the docs.

Reply:
- https://docs.python.org/2.7/library/base64.html
The decoded string is returned. A TypeError is raised if s is incorrectly padded. Characters that are neither in the normal base-64 alphabet nor the alternative alphabet are discarded prior to the padding check.

- https://docs.python.org/3/library/base64.html
If validate is False (the default), characters that are neither in the normal base-64 alphabet nor the alternative alphabet are discarded prior to the padding check. If validate is True, these non-alphabet characters in the input result in a binascii.Error.

Looks like it rasise TypeError?

Reply:
characters being discarded is the common default, so both behave the same.

Reply:
hm, sure. looks like I somehow confused when testing.